webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/score/ScoreService.scala

```diff
@@ -158,9 +158,10 @@ class ScoreServiceImpl(
   }
   def cleanScore(name: String)(implicit qc: QueryContext): IOResult[Unit] = {
     for {
-      _         <- cache.update(_.map { case (id, gscore) => (id, gscore.copy(details = gscore.details.filterNot(_.scoreId == name))) })
+      newGlobal         <- cache.updateAndGet(_.map { case (id, gscore) => (id, gscore.copy(details = gscore.details.filterNot(_.scoreId == name))) })
+      _         <- ZIO.foreach(newGlobal){ case (id, gscore) => globalScoreRepository.save(id, gscore)}
       newScores <- scoreCache.updateAndGet(_.map { case (id, scores) => (id, scores.filterNot(_.scoreId == name)) })
-      _         <- update(newScores)
```

Comment:
I forgot that update keeps value for score that are not present in "newScore", and thought it would clean missing score.
I replaced by adding two queries to remove score directly